plugins/groovy/test/org/jetbrains/plugins/groovy/lang/highlighting/GrResultOfAssignmentUsedTest.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2000-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.plugins.groovy.lang.highlighting
+
+import com.intellij.codeInspection.InspectionProfileEntry
+import org.jetbrains.plugins.groovy.codeInspection.assignment.GroovyResultOfAssignmentUsedInspection
+
+class GrResultOfAssignmentUsedTest extends GrHighlightingTestBase {
+  @Override
+  InspectionProfileEntry[] getCustomInspections() {
+    [new GroovyResultOfAssignmentUsedInspection()]
+  }
+
+  final warnStart = '<warning descr="Result of assignment expression used">'
+  final warnEnd = '</warning>'
+
+  void testUsedVar() {
+    testHighlighting """
+      def foo(xxx) {
+        if ((${warnStart}xxx = 5${warnEnd}) || xxx) {
+          ${warnStart}xxx=4${warnEnd}
+        }
+      }
+
+      def foxo(doo) {
+        def xxx = 'asdf'
+        if (!doo) {
+          println xxx
+          ${warnStart}xxx=5${warnEnd}
+        }
+      }
+
+      def bar(xxx) {
+        print ((${warnStart}xxx=5${warnEnd})?:xxx)
+      }
+
+      def a(xxx) {
+        if (2 && (${warnStart}xxx=5${warnEnd})) {
+          xxx
+        }
+        else {
+          null
+        }
+      }
+    """
+  }
+
+  void testResultOfAssignmentUsed() {
+    testHighlighting """
+      def a() {
+        ${warnStart}b = 1${warnEnd}
```

Comment:
hmmm, on a second thought ... what's the point of a warning here, I don't think it's confusing. Lemme' edit it quickly :)